server/segment_rpc.go

```diff
@@ -37,120 +38,120 @@ var httpClient = &http.Client{
 	Timeout:   HTTPTimeout,
 }
 
-func (h *lphttp) ServeSegment(w http.ResponseWriter, r *http.Request) {
-	orch := h.orchestrator
-
-	payment, err := getPayment(r.Header.Get(PaymentHeader))
-	if err != nil {
-		glog.Error("Could not parse payment")
-		http.Error(w, err.Error(), http.StatusPaymentRequired)
-		return
-	}
+func ServeSegment(orch Orchestrator) http.Handler {
+	return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
```

Comment:
Is there a way we can avoid the extra level of indentation here? It's partially stylistic, but it's a little easier to follow the context and deeper indentation is typically one sign of code smell. For reviewability here, the diff is also a bit hard to follow but I know that's harder to control (but avoiding the indentation would make the diff much cleaner overall).

Although we use this pattern of handler-within-a-closure elsewhere (eg, mediaserver.go) it's still something I'd prefer not to promulgate if we can help it.

Reply:
@j0sh Updated so `ServeSegment` no longer returns `http.Handler` and instead added a test helper that wraps `ServeSegment` in 0b1fda. WDYT?